src/screen.cpp

```diff
@@ -1224,15 +1225,18 @@ void s_reset(screen_t *s, screen_reset_mode_t mode) {
                 abandon_line_string.append(
                     str2wcstring(tparm(exit_attribute_mode)));  // normal text ANSI escape sequence
             }
-            abandon_line_string.append(screen_width - non_space_width, L' ');
+            int newline_glitch_width = term_has_xn ? 0 : 1;
+            abandon_line_string.append(screen_width - non_space_width - newline_glitch_width, L' ');
```

Comment:
Done. I changed the `>=` that protects this code to `>` although if you'd prefer I could change it to

```
if (screen_width >= non_space_width + 1) {
```